aws/resource_aws_network_acl_rule.go

```diff
@@ -210,8 +235,8 @@ func resourceAwsNetworkAclRuleRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("ipv6_cidr_block", resp.Ipv6CidrBlock)
 	d.Set("egress", resp.Egress)
 	if resp.IcmpTypeCode != nil {
-		d.Set("icmp_code", resp.IcmpTypeCode.Code)
-		d.Set("icmp_type", resp.IcmpTypeCode.Type)
+		d.Set("icmp_code", strconv.FormatInt(*resp.IcmpTypeCode.Code, 10))
+		d.Set("icmp_type", strconv.FormatInt(*resp.IcmpTypeCode.Type, 10))
```

Comment:
To prevent potential panics, we should use the [AWS Go SDK provided conversion functions](https://docs.aws.amazon.com/sdk-for-go/api/aws/), e.g. `aws.StringValue()`, here:

```suggestion
		d.Set("icmp_code", strconv.FormatInt(aws.StringValue(resp.IcmpTypeCode.Code), 10))
		d.Set("icmp_type", strconv.FormatInt(aws.StringValue(resp.IcmpTypeCode.Type), 10))
```